rs/crypto/test_utils/vetkd/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use ic_crypto_internal_bls12_381_type::{G1Affine, G2Affine, Scalar};
+use ic_crypto_internal_bls12_381_vetkd::{
+    DerivationContext, EncryptedKey, EncryptedKeyShare, TransportPublicKey,
+};
+
+pub fn dummy_transport_public_key() -> [u8; 48] {
+    G1Affine::generator().serialize()
+}
+
+pub struct PrivateKey {
+    secret_key: Scalar,
+    public_point: G2Affine,
+    pk_bytes: Vec<u8>,
+}
+
+impl PrivateKey {
+    pub fn generate(seed: &[u8]) -> Self {
+        let secret_key = Scalar::hash(seed, b"ic-crypto-vetkd-test-utils-generate-test-key");
+        Self::from_scalar(secret_key)
+    }
+
+    fn from_scalar(secret_key: Scalar) -> Self {
+        let public_point = G2Affine::from(G2Affine::generator() * &secret_key);
+        let pk_bytes = public_point.serialize().to_vec();
+        Self {
+            secret_key,
+            public_point,
+            pk_bytes,
+        }
+    }
+
+    pub fn public_key_bytes(&self) -> Vec<u8> {
+        self.pk_bytes.clone()
+    }
+
+    pub fn vetkd_protocol(
+        &self,
+        canister_id: &[u8],
+        context: &[Vec<u8>],
+        input: &[u8],
+        tpk: &[u8],
+    ) -> Vec<u8> {
+        let dc = if context.is_empty() {
+            DerivationContext::new(canister_id, &[])
+        } else {
+            DerivationContext::new(canister_id, &context[0])
+        };
+
+        let tpk =
+            TransportPublicKey::deserialize(tpk).expect("Failed to deserialize TransportPublicKey");
+
+        let mut rng = rand::thread_rng();
+
+        let eks = EncryptedKeyShare::create(
+            &mut rng,
+            &self.public_point,
+            &self.secret_key,
+            &tpk,
+            &dc,
+            input,
+        );
+
+        let ek = EncryptedKey::combine_all(&[(1, eks)], 1, &self.public_point, &tpk, &dc, input)
```

Comment:
Hm...in the chainkey-testing-canister ([here](https://github.com/dfinity/chainkey-testing-canister/blob/ae280abe19c2bb082b9d1660f280e75d00c96543/src/vetkd.rs#L105)) and in the dummy system API impl in the examples repo ([here](https://github.com/dfinity/examples/blob/375aceacde7708f84615e73a275565036e3098c7/rust/vetkd/src/system_api/src/lib.rs#L82)) we used node index 0...now I'm wondering if that is incorrect.

Side note: now that we have this test utility, we could make use of it at least in the chainkey-testing-canister (as the chainkey-testing-canister will stay; while the dummy system API impl will disappear hopefully soon).

Reply:
I would have to double check on this, to be honest. We may be assuming nodes are [0...n) and always +1 to translate from the node index to the value.

Reply:
There should be a test within this test-utils crate that the `EncryptedKey` we generate here can be decrypted/verified

Reply:
Confirmed, node indexes should be 0..n. I fixed this and added a test that the generated EncryptedVetKey can be decrypted.